src/java/picard/util/JavascriptFilter.java

```diff
@@ -0,0 +1,215 @@
+	}
+
+	/** returns key used for header binding */
+	public String getHeaderKey() {
+		return HEADER_KEY;
+	}
+
+	/** returns key used for record binding */
+	public abstract String getRecordKey();
+
+	/** Evaluates this predicate on the given argument. */
+	protected boolean accept(final TYPE record) {
+		try {
+			this.bindings.put(getRecordKey(), record);
+			Object result = this.script.eval(bindings);
+			if (result == null)
+				return false;
+			;
+			if (result instanceof Boolean) {
+				if (Boolean.FALSE.equals(result))
+					return false;
+			} else if (result instanceof Number) {
+				if (((Number) result).intValue() != 1)
+					return false;
+			} else {
+				return false;
+			}
+			return true;
+		} catch (ScriptException err) {
+			throw new RuntimeException(err);
+		}
+	}
+
+	/** creates the Javascript manager, read and compile the scriptFile */
+	protected void initialize(final File scriptFile) {
+		FileReader scriptReader = null;
+		final ScriptEngineManager manager = new ScriptEngineManager();
+		this.engine = manager.getEngineByName("js");
+		if (this.engine == null) {
+			throw new RuntimeException("The embedded 'javascript' engine is not available in java. "
+			        + "Do you use the SUN/Oracle Java Runtime ?");
+		}
+		if (scriptFile == null)
+			throw new NullPointerException("missing javascript file.");
+		IOUtil.assertFileIsReadable(scriptFile);
+		try {
+			final Compilable compilingEngine = (Compilable) this.engine;
+			scriptReader = new FileReader(scriptFile);
+			this.script = compilingEngine.compile(scriptReader);
+		} catch (IOException err) {
+			throw new RuntimeException("I/O exception in " + scriptFile, err);
+		} catch (ScriptException err) {
+			throw new RuntimeException("Script error in " + scriptFile, err);
+		} finally {
+			CloserUtil.close(scriptReader);
+		}
+
+	}
+
+	/**
+	 * creates a SamRecordFilter from the given SAMFileHeader and the user
+	 * javascript file
+	 */
+	static public SamRecordFilter makeSamFilter(final SAMFileHeader header, final File scriptFile) {
+		final JavascriptSamRecordFilter r = new JavascriptSamRecordFilter(header);
+		r.initialize(scriptFile);
+		return r;
+	}
+
+	/**
+	 * creates a VariantFilter from the given VCFHeader and the user javascript
+	 * file
+	 */
+	static public VariantFilter makeVariantContextFilter(final VCFHeader header, final File scriptFile) {
+		final JavascriptVariantContextFilter r = new JavascriptVariantContextFilter(header, scriptFile);
+		r.initialize(scriptFile);
+		return r;
+	}
+
+	/** implementation of a JavascriptSamRecordFilter as a SamRecordFilter */
+	private static class JavascriptSamRecordFilter extends JavascriptFilter<SAMFileHeader, SAMRecord>
+	        implements SamRecordFilter {
+		public JavascriptSamRecordFilter(final SAMFileHeader header) {
+			super(header);
+		}
+
+		@Override
+		public boolean filterOut(final SAMRecord first, final SAMRecord second) {
+			return filterOut(first) || filterOut(second);
```

Comment:
due to the reverse logic, this will exclude read-pairs if either one of them gets false from the javascript. This should either be an option, or not allowed, or clearly stated, since it isn't obvious from reading.